zeppelin-user-impersonation[3016].md

```diff
@@ -0,0 +1,152 @@
+role2 = * </br>
+role3 = * </br>
+admin = * </br>
+[urls] </br>
+###This section is used for url-based security. </br>
+###You can secure interpreter, configuration and credential information by urls. Comment or uncomment the below urls that you want to hide. </br>
+###anon means the access is anonymous. </br>
+###authc means Form based Auth Security </br>
+###To enfore security, comment the line below and uncomment the next one </br>
+/api/version = anon </br>
+#/api/interpreter/** = authc, roles[admin] </br>
+#/api/configurations/** = authc, roles[admin] </br>
+#/api/credential/** = authc, roles[admin] </br>
+#/** = anon </br>
+/** = authc </br>
+</i>
+
+<b>Below are the configurations required from the environment:</b> </br>
+•	On the Zeppelin server node make sure to add the below to <i>/etc/sudoers</i>: </br>
+With the root user: </br>
+
+`$visudo` </br>
+`zeppelin        ALL=(ALL)       NOPASSWD: ALL ` </br> 
+
+If this entry is missing in the /etc/sudoers then the interpreter fails with the below error: </br>
+
+`org.apache.zeppelin.interpreter.InterpreterException: Permission denied, please try again.`</br>
+`Permission denied, please try again.` </br>
+`Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).`</br> 
+</br>
+
+•	Log folder (usually /var/log/zeppelin or /var/log/hadoop/zeppelin) of zeppelin should be of 777 instead of 755 if not the Spark interpreter keep failing with below error as the impersonated user won’t able to write to the logs folder:
+
+`java.lang.NullPointerException`</br>
+	`at org.apache.zeppelin.spark.Utils.invokeMethod(Utils.java:38)`</br>
+	`at org.apache.zeppelin.spark.Utils.invokeMethod(Utils.java:33)` </br>
+
+</br>
+•	If you are on a HDP platform this can’t be achieved from the CLI which needs a Ambari Zeppelin code level change </br>
+
+`$ vi /var/lib/ambari-server/resources/common-services/ZEPPELIN/0.6.0.2.5/package/scripts/master.py` </br>
+
+#### Find the below: </br>
+
+```
+def create_zeppelin_log_dir(self, env):
+    import params
+    env.set_params(params)
+    Directory([params.zeppelin_log_dir],
+              owner=params.zeppelin_user,
+              group=params.zeppelin_group,
+              cd_access="a",
+              create_parents=True,
+              mode=0755
+              )
+  ```          
+##### Change this to:
+
+```
+def create_zeppelin_log_dir(self, env):
+    import params
+    env.set_params(params)
+    Directory([params.zeppelin_log_dir],
+              owner=params.zeppelin_user,
+              group=params.zeppelin_group,
+              cd_access="a",
+              create_parents=True,
+              mode=0777
+              )
+```
+##### Please note this requires the stop and start of AMBARI-SERVER. </br>
+Configurations required for Zeppelin User Impersonation: </br>
+
+•	Add the proxy settings for zeppelin user under core-site.xml and restart the required services (HDFS, Mapreduce2 and YARN) </br>
+
+`hadoop.proxyuser.zeppelin.hosts=*` </br>
```

Comment:
It depends on which user zeppelin is launcher, if it is launcher via user, then it is `hadoop.proxyuser.zeppelin.hosts`